extensions/hazelcast-client/deployment/src/main/java/io/quarkus/hazelcast/client/deployment/HazelcastClientProcessor.java

```diff
@@ -0,0 +1,237 @@
+import com.hazelcast.config.replacer.EncryptionReplacer;
+import com.hazelcast.config.replacer.PropertyReplacer;
+import com.hazelcast.config.replacer.spi.ConfigReplacer;
+import com.hazelcast.gcp.GcpDiscoveryStrategy;
+import com.hazelcast.gcp.GcpDiscoveryStrategyFactory;
+import com.hazelcast.nio.serialization.DataSerializable;
+import com.hazelcast.nio.serialization.DataSerializableFactory;
+import com.hazelcast.nio.serialization.PortableFactory;
+import com.hazelcast.nio.serialization.Serializer;
+import com.hazelcast.nio.ssl.BasicSSLContextFactory;
+import com.hazelcast.spi.discovery.DiscoveryStrategy;
+import com.hazelcast.spi.discovery.DiscoveryStrategyFactory;
+import com.hazelcast.spi.discovery.NodeFilter;
+import com.hazelcast.spi.discovery.multicast.MulticastDiscoveryStrategy;
+import com.hazelcast.util.ICMPHelper;
+
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.ExtensionSslNativeSupportBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.JniBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.NativeImageResourceBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.NativeImageResourceBundleBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveHierarchyBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.RuntimeReinitializedClassBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ServiceProviderBuildItem;
+import io.quarkus.deployment.util.ServiceUtil;
+import io.quarkus.hazelcast.client.runtime.HazelcastClientBytecodeRecorder;
+import io.quarkus.hazelcast.client.runtime.HazelcastClientConfig;
+import io.quarkus.hazelcast.client.runtime.HazelcastClientProducer;
+import io.quarkus.jaxb.deployment.JaxbFileRootBuildItem;
+
+class HazelcastClientProcessor {
+
+    BuildProducer<ReflectiveClassBuildItem> reflectiveClasses;
+    BuildProducer<NativeImageResourceBuildItem> resources;
+    BuildProducer<ReflectiveHierarchyBuildItem> reflectiveClassHierarchies;
+    BuildProducer<NativeImageResourceBundleBuildItem> bundles;
+    BuildProducer<RuntimeReinitializedClassBuildItem> reinitializedClasses;
+    BuildProducer<ServiceProviderBuildItem> services;
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FeatureBuildItem.HAZELCAST_CLIENT);
+    }
+
+    @BuildStep
+    void enableSSL(BuildProducer<ExtensionSslNativeSupportBuildItem> ssl) {
+        ssl.produce(new ExtensionSslNativeSupportBuildItem(FeatureBuildItem.HAZELCAST_CLIENT));
+    }
+
+    @BuildStep
+    void enableJNI(BuildProducer<JniBuildItem> jni) {
+        jni.produce(new JniBuildItem());
```

Comment:
You can drop that one, it's not needed anymore in master as 19.3 has JNI always enabled.